frontend/src/app/json-editor/page.tsx

```diff
@@ -0,0 +1,603 @@
+"use client";
+
+import { Alert, AlertDescription, AlertTitle } from "@/components/ui/alert";
+import { Button } from "@/components/ui/button";
+import { Calendar } from "@/components/ui/calendar";
+import { Input } from "@/components/ui/input";
+import { Popover, PopoverContent, PopoverTrigger } from "@/components/ui/popover";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import { Switch } from "@/components/ui/switch";
+import { Textarea } from "@/components/ui/textarea";
+import { fetchCategories } from "@/lib/data";
+import { Category } from "@/lib/types";
+import { cn } from "@/lib/utils";
+import { CalendarIcon, Check, Clipboard, PlusCircle, Trash2 } from "lucide-react";
+import { useEffect, useState } from "react";
+import { toast } from "sonner";
+import { z } from "zod";
+import { format } from "date-fns";
+import { Label } from "@/components/ui/label";
+import { AlertColors } from "@/config/siteConfig";
+
+const scriptSchema = z.object({
+  name: z.string().min(1),
+  slug: z.string().min(1),
+  categories: z.array(z.number()),
+  date_created: z.string().regex(/^\d{4}-\d{2}-\d{2}$/),
+  type: z.enum(["vm", "ct", "misc"]),
+  updateable: z.boolean(),
+  privileged: z.boolean(),
+  interface_port: z.number().nullable(),
+  documentation: z.string().nullable(),
+  website: z.string().url().nullable(),
+  logo: z.string().url().nullable(),
+  description: z.string().min(1),
+  install_methods: z.array(
+    z.object({
+      type: z.enum(["default", "alpine"]),
+      script: z.string().min(1),
+      resources: z.object({
+        cpu: z.number().nullable(),
+        ram: z.number().nullable(),
+        hdd: z.number().nullable(),
+        os: z.string().nullable(),
+        version: z.number().nullable(),
+      }),
+    }),
+  ),
+  default_credentials: z.object({
+    username: z.string().nullable(),
+    password: z.string().nullable(),
+  }),
+  notes: z.array(
+    z.object({
+      text: z.string().min(1),
+      type: z.string().min(1),
+    }),
+  ),
+});
+
+type Script = z.infer<typeof scriptSchema>;
+
+export default function JSONGenerator() {
+  const [script, setScript] = useState<Script>({
+    name: "",
+    slug: "",
+    categories: [],
+    date_created: "",
+    type: "vm",
+    updateable: false,
+    privileged: false,
+    interface_port: null,
+    documentation: null,
+    website: null,
+    logo: null,
+    description: "",
+    install_methods: [],
+    default_credentials: {
+      username: null,
+      password: null,
+    },
+    notes: [],
+  });
+  const [isCopied, setIsCopied] = useState(false);
+
+  const [isValid, setIsValid] = useState(false);
+  const [categories, setCategories] = useState<Category[]>([]);
+
+  useEffect(() => {
+    fetchCategories()
```

Comment:
Perhaps we need another fetch which does not include all the scripts, since they are not needed here? Can do it later though, just an optimization

Reply:
i did it in advance, since i wanted to add a feature to edit scripts